### ISSUE TITLE
Composite uniqueKey not working

### ISSUE THREAD

**Issue by maxpain** [join-monster/join-monster#58]
Example: user has a friends list (join), when user has no friends, gql server still returns object with empty (null) fields in `friends` array. 

Friends postgres schema (user_id and friend_id — primary keys):
```sql
CREATE TABLE "public"."friends" (
	"user_id" int4 NOT NULL,
	"friend_id" int4 NOT NULL,
	"date" timestamp(6) NOT NULL DEFAULT now(),
	"state" "public"."friends_state" NOT NULL
)
```

This query:
```graphql
query {
  user {
    id
    nickName
    friends {
      date
      user {
        id
      }
    }
  }
}
```
returns this response:
```javascript
{
  "data": {
    "user": {
      "id": 251586469,
      "nickName": "MAXPAIN",
      "friends": [
        {
          "date": null,
          "user": null
        }
      ]
    }
  }
}
```

GraphQL schema and resolvers:

`User.js`
```javascript
import {
	GraphQLObjectType,
	GraphQLList,
	GraphQLString,
	GraphQLInt
} from 'graphql'

import Friend from '../friend'

export default new GraphQLObjectType({
	description: 'A user type',
	name: 'User',
	sqlTable: 'users',
	uniqueKey: 'id',
	fields: () => ({
		id: {
			type: GraphQLInt
		},
		nickName: {
			type: GraphQLString,
			sqlColumn: 'nick_name'
		},
		friends: {
			description: 'A list of user friends',
			type: new GraphQLList(Friend),
			sqlJoin: (userTable, friendTable) => `${friendTable}.user_id = ${userTable}.id`
		}
	})
})
```

`Friend.js`
```javascript
import {
	GraphQLObjectType,
	GraphQLString
} from 'graphql'

import User from '../user'

export default new GraphQLObjectType({
	description: 'A friend type',
	name: 'Friend',
	sqlTable: 'friends',
	uniqueKey: ['user_id', 'friend_id'],
	fields: () => ({
		user: {
			type: User,
			sqlJoin: (friendTable, userTable) => `${userTable}.id = ${friendTable}.friend_id`
		},
		date: {
			type: GraphQLString
		}
	})
})
```

Dump:
```
  API Tue, 24 Jan 2017 15:52:53 GMT join-monster
  API SQL_AST
  API  { type: 'table',
  API   name: 'users',
  API   as: 'user',
  API   fieldName: 'user',
  API   grabMany: false,
  API   where: [Function: where],
  API   children:
  API    [ { type: 'column', name: 'id', fieldName: 'id', as: 'id' },
  API      { type: 'column', name: 'id', fieldName: 'id', as: 'id' },
  API      { type: 'column',
  API        name: 'nick_name',
  API        fieldName: 'nickName',
  API        as: 'nick_name' },
  API      { type: 'table',
  API        name: 'friends',
  API        as: 'friends',
  API        fieldName: 'friends',
  API        grabMany: true,
  API        sqlJoin: [Function: sqlJoin],
  API        children:
  API         [ { type: 'composite',
  API             name: [ 'user_id', 'friend_id' ],
  API             fieldName: 'user_id#friend_id',
  API             as: 'user_id#friend_id' },
  API           { type: 'column', name: 'date', fieldName: 'date', as: 'date' },
  API           { type: 'table',
  API             name: 'users',
  API             as: 'user$',
  API             fieldName: 'user',
  API             grabMany: false,
  API             sqlJoin: [Function: sqlJoin],
  API             children:
  API              [ { type: 'column', name: 'id', fieldName: 'id', as: 'id' },
  API                { type: 'column', name: 'id', fieldName: 'id', as: 'id' },
  API                { type: 'columnDeps', names: {} } ] },
  API           { type: 'columnDeps', names: {} } ] },
  API      { type: 'columnDeps', names: {} } ] }
  API  +176ms
  API Tue, 24 Jan 2017 15:52:53 GMT join-monster
  API SQL
  API  SELECT
  API   "user"."id" AS "id",
  API   "user"."nick_name" AS "nick_name",
  API   CONCAT("friends"."user_id", "friends"."friend_id") AS "friends__user_id#friend_id",
  API   "friends"."date" AS "friends__date",
  API   "user$"."id" AS "friends__user$__id"
  API FROM users AS "user"
  API LEFT JOIN friends AS "friends" ON "friends".user_id = "user".id
  API LEFT JOIN users AS "user$" ON "user$".id = "friends".friend_id
  API WHERE "user".id = 251586469
  API  +4ms
  API Tue, 24 Jan 2017 15:52:53 GMT join-monster
  API SHAPE_DEFINITION
  API  { id: 'id',
  API   nickName: 'nick_name',
  API   friends:
  API    [ { 'user_id#friend_id': 'friends__user_id#friend_id',
  API        date: 'friends__date',
  API        user: { id: 'friends__user$__id' } } ] }
  API  +1ms
  API Tue, 24 Jan 2017 15:52:53 GMT join-monster
  API RAW DATA
  API  [ anonymous {
  API     id: 251586469,
  API     nick_name: 'MAXPAIN',
  API     'friends__user_id#friend_id': '',
  API     friends__date: null,
  API     'friends__user$__id': null } ]
  API  +60ms
```

**Comment by acarl005** [comment]
Hm. The SQL query appears to be correct. Yet the raw data lacks any joined friendships. Are you certain that this user has any friends?

If you paste the SQL query into the DBMS directly, what result do you get? Is it what you expect?

If so, what happens if you switch the composite key to just a single column? Does it produce any friendships at all?

**Comment by maxpain** [comment]
If just a single column using as uniqueKey, then all works fine.

**Comment by maxpain** [comment]
I want `friends: []` if user has no friends.

**Comment by acarl005** [comment]
I see. So the problem is that you have an object with nulls in the array rather than an empty array. That's certainly buggy behavior. I'll look into this.

**Comment by maxpain** [comment]
@acarl005 Thx

**Comment by acarl005** [comment]
Which database and which version are you using?

**Comment by maxpain** [comment]
@acarl005 
postgresql: 9.6
join-monster: 0.9.4

**Comment by acarl005** [comment]
The problem was with the way `CONCAT` works in Postgres when passed two `NULL` values.

The intended behavior was that `CONCAT(NULL, NULL)` would return `NULL`, as it does in MySQL. But Postgres returns the empty string instead. The empty string for the value of a key is not treated as an absence during the hydration process. It therefore created an object when it should not have.

This was fixed in 4ab80f5 and published in **v0.9.5**.

Thanks for reporting the bug and providing all the helpful info.

**Comment by maxpain** [comment]
@acarl005 Thank you.
